app/resources/report/api.py

```diff
@@ -12,18 +13,22 @@
     extract_span_from_kwargs,
     trace,
 )
+from sqlalchemy.orm import joinedload
 
+from app.extensions import db
 from app.libs.resource import ResourceHandler
 from app.resources.product.models import Product
 from app.resources.sli import sources
 from app.resources.sli.models import Indicator
 from app.resources.slo.models import Objective
+from app.resources.target.models import Target
 
 REPORT_TYPES = ('weekly', 'monthly', 'quarterly')
 
 
-def get_report_params(report_type,) -> Tuple[sources.DatetimeRange, sources.Resolution]:
-    to_dt = datetime.utcnow()
+def get_report_params(
+    report_type, to_dt=None,
+) -> Tuple[sources.DatetimeRange, sources.Resolution]:
     if report_type == "weekly":
         from_dt = to_dt - relativedelta(days=7)
```

Comment:
what happens here if `to_dt` is `None`?

Reply:
I just forgot about this default value. Caller is responsible for `to_dt` to be not None and code that uses this function always fills a default.